feat- Translation of enum cases

andrefajardo opened the pull request:
Implementation of Portuguese translation for enum cases

<Resp: José André - Ref #Card-5>

andrefajardo committed:
feat- Translation of enum cases

Implementation of Portuguese translation for enum cases

<Resp: José André - Ref #Card-5>

lucascarnu commented on src/enums/EnrollmentStatus.java:
ACTIVE = ATIVO
LOCKED = TRANCADO ¿?
GRADUATED = DIPLOMADO

Eu acho que seria melhor tirar o método que faz a tradução e diretamente no código do enum colocar as palavras em português. Exemplo:
package enums;

public enum EnrollmentStatus {
    ATIVO(1),
    TRANCADO(2),
    DIPLOMADO(3);

andrefajardo replied:
Boa tarde, Lucas.
Sua observação é totalmente pertinente e o ajuste foi providenciado, apenas com a alteração da nomenclatura de  DIPLOMADO para GRADUADO.

andrefajardo committed:
Fix feat- Translation of enum cases

Implementation of correct Portuguese translation for enum EnrollmentStatus cases.

<Resp: José André - Ref #Card-5>

micheldrv commented on src/enums/EmployeeLevel.java:
```diff
@@ -27,4 +27,19 @@ public static EmployeeLevel getStatusNameFromIndex(int num) {
+            }
+        }
+        return null;
+    }
```
Esse código não tá funcionando como o esperado. È sempre bom testar pra ver se a funcionalidade está como desejada.

Essa função não está utilizando o parâmetro `num`. Está iterando por todos os valores do enum, e, se for igual a 1, retorna "INICIANTE". Por isso, no teste que fiz, essa função sempre retorna "INICIANTE" independente do valor `num` inserido. O mesmo vale pro método da EnrollmentStatus.

Além do mais, a minha sugestão é que seria melhor a função receber um valor do próprio enum, ao invés de int. Assim poderiam chamar, por exemplo, `EmployeeLevel.getTranslatedStatusNameFromIndex(EmployeeLevel.BEGINNER);`

Também acho que não deveria retornar a palavra toda em maiúscula. Pra ser printada para o usuário, acho que seria melhor retornar "Iniciante", "Experiente", etc.

Outra sugestão é que seria bom usar um switch nessa lógica, pelo menos faz mais sentido pra mim do que else ifs.

andrefajardo replied:
Perfeito, Michel. Vou refazer e providenciar as correções.
